### PR TITLE
access_context_t work

### PR DESCRIPTION
**Pull request by tklengyel**
This series converts some functions to use access_context_t instead of the current vaddr and PID input.  With access_context_t it these functions become more flexible as the user will be able to pass a DTB in directly. This involves changes to the public API.

As part of this change the symcache and rvachache has been updated to use the DTB as unique identifier instead of the PID.

### CODE REVIEW

**Comment by tklengyel** [comment]
Retest this please.

**Comment by tklengyel** [comment]
Retest this please.

**Comment by smaresca** [comment]
+1, I have tested + used this branch already. Good to go.

File: libvmi/libvmi.h [hunk 1]
```diff
@@ -754,6 +750,20 @@ status_t vmi_read_addr(
...
+ * supported. The returned value must be freed by the caller.
+ *
+ * @param[in] vmi LibVMI instance
+ * @param[in] ctx Access context
+ * @return String read from memory or NULL on error; this function
+ *         will set the encoding field.
+ */
+unicode_string_t *vmi_read_unicode_str(
+    vmi_instance_t vmi,
     access_context_t *ctx);
```
Comment (tklengyel) [inline libvmi/libvmi.h]: This should probably be a const as well..

**Pull request merged**